### ISSUE TITLE
Manual pages, user guide aren't installed

### ISSUE THREAD

**Issue by hseg** [haskell/ghcup-hs#990]
Contra the description in https://www.haskell.org/ghcup/guide/#manpages, a bare `ghcup install ghc 9.8.1` does not install manpages or the user guide.
Instead, `$(ghcup whereis basedir)/ghc/9.8.1/share/doc/ghc-9.8.1/` contains the Haddocks for the libraries bundled with GHC. (similarly, the GHC API docs aren't installed)

**Comment by hasufell** [comment]
What do you mean with "install"?

**Comment by hseg** [comment]
On Sat, Jan 27, 2024 at 06:02:02PM -0800, Julian Ospald wrote:
> What do you mean with "install"?

Was expecting (especially given the descriptions in the user guide and the
presence of `~/.local/share/ghcup/share/doc/ghc-9.8.1/html/libraries/`) for a
copy of each version's user guide to be installed to `.../html/users_guide/` and
for the manual to be installed to eg `.../share/doc/ghc-9.8.1/man/`.
How else is `man-db` supposed to know what to display when invoking `man ghc`?

**Comment by hasufell** [comment]
Are you sure there is no manpage? Did you follow the instructions here: https://www.haskell.org/ghcup/guide/#manpages

This did work for me. I can try later. Please don't assume specific locations of files.

Whether the user guide is available or not may depend on which bindist exactly was installed. Which one was it?

**Comment by hasufell** [comment]
Also, I don't recommend enabling XDG support.

**Comment by hseg** [comment]
On Sun, Jan 28, 2024 at 05:31:26AM -0800, Julian Ospald wrote:
> Are you sure there is no manpage? Did you follow the instructions here: https://www.haskell.org/ghcup/guide/#manpages
> 
> This did work for me. I can try later. Please don't assume specific locations of files.

I checked every directory listed in `ghcup debug-info` -- excluding extensions
that are probably not manpages, I couldn't find any file that could be a
manpage.

> Whether the user guide is available or not may depend on which bindist exactly was installed. Which one was it?

Am using the ghc 9.8.1 installation, don't know what bindist that corresponds
to. But neither do I see any file matching `gz` in
`ghc-9.4.8-x86_64-fedora27-linux.tar.xz`, which is the recommended bindist, and
indeed installing it again shows an absence of possible manpage files.
Though it ran out of space on my computer and aborted, so it might not be
indicative of anything, still I recall trying this earlier and not getting
manpages in other bindists.

**Comment by hasufell** [comment]
manpages work fine here:

```
$ file ~/.ghcup/share
/home/hasufell/.ghcup/share: symbolic link to ./ghc/9.4.8/share
$ file ~/.ghcup/share/man/man1/ghc.1
/home/hasufell/.ghcup/share/man/man1/ghc.1: troff or preprocessor input, Unicode text, UTF-8 text, with very long lines (7359)
```

Maybe it doesn't work with XDG, but again: I don't recommend XDG.

----

Wrt the user-guide, it seems it's sometimes just a PDF and sometimes also HTML, compare:

```
$ file ~/.ghcup/ghc/9.8.1/share/doc/ghc-9.8.1/html/users_guide/index.html
/home/hasufell/.ghcup/ghc/9.8.1/share/doc/ghc-9.8.1/html/users_guide/index.html: HTML document, Unicode text, UTF-8 text
$ find ~/.ghcup/ghc/9.4.8/share/doc/ghc-9.4.8/users_guide/
/home/hasufell/.ghcup/ghc/9.4.8/share/doc/ghc-9.4.8/users_guide/
/home/hasufell/.ghcup/ghc/9.4.8/share/doc/ghc-9.4.8/users_guide/build-man
/home/hasufell/.ghcup/ghc/9.4.8/share/doc/ghc-9.4.8/users_guide/build-man/ghc.1
$ file ~/.ghcup/ghc/9.4.8/share/doc/ghc-9.4.8/users_guide.pdf
/home/hasufell/.ghcup/ghc/9.4.8/share/doc/ghc-9.4.8/users_guide.pdf: PDF document, version 1.5 (zip deflate encoded)
```

----

So all in all: we install everything that's in the bindist. If it has manpages, they are installed. If it has html userguide, it's gonna be installed as well.

**Comment by hseg** [comment]
Investigating the contents of
`https://downloads.haskell.org/~ghc/latest/ghc-9.8.1-x86_64-rocky8-linux.tar.xz`.
I should be getting the manpages -- suspect this is my configurations not
playing nicely with Hadrian. Will see what I can find on this.

**Comment by hseg** [comment]
Sorry, didn't find the time since to check what's going on.
Tested again with:
> $ for i in {$XDG_CONFIG_HOME,$XDG_CACHE_HOME,$XDG_DATA_HOME}/ghcup; do mv "$i" "$i"-bak; done; mkdir ~/.local/bin/ghcup-bak; mv ~/.local/bin/{cabal,ghc,haddock,hp,hsc,runghc,runhaskell}* ~/.local/bin/ghcup-bak
> $ unset GHCUP_USE_XDG_DIRS
> $ ghcup nuke # Note -- doesn't remove the ghcup binary itself since it is managed by my package manager
> $ ghcup install ghc recommended
> $ file ~/.ghcup/share ~/.ghcup/share/man/man1/ghc.1
> /home/gesh/.ghcup/share:                cannot open `/home/gesh/.ghcup/share' (No such file or directory)
> /home/gesh/.ghcup/share/man/man1/ghc.1: cannot open `/home/gesh/.ghcup/share/man/man1/ghc.1' (No such file or directory)

**Comment by hseg** [comment]
Building off HEAD, checking if that resolves things...

**Comment by hseg** [comment]
No dice -- even without using XDG dirs and on HEAD, still am not seeing any man directory in ~/.ghcup/ghc/9.4.8/share/

**Comment by hasufell** [comment]
```
$ ls .ghcup/ghc/9.4.8/share/man/man1/ghc.1
.ghcup/ghc/9.4.8/share/man/man1/ghc.1
```

**Comment by hseg** [comment]
ls: cannot access '/home/gesh/.ghcup/ghc/9.4.8/share/man/man1/ghc.1': No such file or directory

**Comment by hasufell** [comment]
Yeah, so the ghc-9.4.8-x86_64-fedora27-linux.tar.xz bindist doesn't seem to have/install manpages. And neither does ghc-9.8.1-x86_64-linux-rocky8.tar.xz.

That's an upstream issue. GHCup may (in the future) build bindists independently, but we don't do that right now. I'm unlikely to invest a lot of time into that right now.

**Comment by hseg** [comment]
This appears to be a similarly-scoped issue to #298

**Comment by hasufell** [comment]
Feel free to open a ticket, but in my experience GHC developers put very little effort in curating past releases.

I don't want to get involved in GHC upstream bindists. I'll rather build them on my own.

**Comment by hseg** [comment]
Of course -- was expecting this request to be applied to releases moving forward.
What would you need from GHC upstream for this? That `make install` also copy over the manpages and userguide?

**Comment by chreekat** [comment]
They should at least be included in the bindists, though I suspect the reason they aren't may be that all the documentation is built in one big glop and it's way too expensive to build all of it for all the different bindists. Not that it's an excuse---just that it might take real work to fix. I've opened https://gitlab.haskell.org/ghc/ghc/-/issues/24465 at any rate.

**Comment by hseg** [comment]
On Thu, Feb 22, 2024 at 05:27:17AM -0800, Bryan Richter wrote:
> They should at least be included in the bindists, though I suspect the reason they aren't may be that all the documentation is built in one big glop and it's way too expensive to build all of it for all the different bindists. Not that it's an excuse---just that it might take real work to fix. I've opened https://gitlab.haskell.org/ghc/ghc/-/issues/24465 at any rate.

Couldn't they share work between the bindists? This seems like it should be a
shared build product.
In any case, thanks for filing this!

**Comment by hasufell** [comment]
https://github.com/haskell/ghcup-metadata/actions/runs/8030282571/job/21937493709

Affected bindists:

* https://downloads.haskell.org/~ghc/9.8.2/ghc-9.8.2-x86_64-unknown-mingw32.tar.xz (missing manpage only)
* https://downloads.haskell.org/~ghc/9.8.2/ghc-9.8.2-x86_64-apple-darwin.tar.xz
* https://downloads.haskell.org/~ghc/9.8.2/ghc-9.8.2-aarch64-apple-darwin.tar.xz
* https://downloads.haskell.org/~ghc/9.8.2/ghc-9.8.2-x86_64-centos7-linux.tar.xz
* https://downloads.haskell.org/~ghc/9.8.2/ghc-9.8.2-x86_64-rocky8-linux.tar.xz
* https://downloads.haskell.org/~ghc/9.8.2/ghc-9.8.2-x86_64-alpine3_12-linux.tar.xz

**Comment by hasufell** [comment]
This has been manually fixed for 9.8.2 on GHCup's side.

I'm considering to fix it for 9.4.8 too.

A CI test has been added here: https://github.com/haskell/ghcup-metadata/commit/80e1d2443f8f08fd9234db4a6eb468abfd162e41

**Comment by hseg** [comment]
On Wed, Feb 28, 2024 at 09:28:33PM -0800, Julian Ospald wrote:
> This has been manually fixed for 9.8.2 on GHCup's side.
> 
> I'm considering to fix it for 9.4.8 too.
> 
> A CI test has been added here: https://github.com/haskell/ghcup-metadata/commit/80e1d2443f8f08fd9234db4a6eb468abfd162e41

Even though I know it's not recommended, thought you'd appreciate knowing that
testing shows this fix for 9.8.2 works with XDG dirs. Need to configure man-db
on my end to pick the manpage up, but that's a me problem. Thank you!
